File: skills/runninghub-api-dev/assets/sdk/python/examples/run_app.py
#!/usr/bin/env python3
from __future__ import annotations

import argparse
import json
import os
from pathlib import Path

from app_presets import APP_PRESETS, build_app_overrides
from quick_create_client import AiAppRunner
from runninghub_client import RunningHubClient


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="运行 RunningHub AI 应用示例")
    parser.add_argument("--app-id", required=True, choices=sorted(APP_PRESETS.keys()))
    parser.add_argument("--file", action="append", default=[], help="本地输入文件，可重复传入")
    parser.add_argument("--prompt", default=None, help="覆盖默认提示词")
    parser.add_argument("--save-json", default=None, help="保存执行结果到 JSON 文件")
    parser.add_argument("--poll-interval", type=int, default=12)
    parser.add_argument("--timeout-seconds", type=int, default=1800)
    return parser.parse_args()


def main() -> int:
    args = parse_args()
    api_key = os.environ.get("RUNNINGHUB_API_KEY")
    if not api_key:
        raise SystemExit("缺少环境变量 RUNNINGHUB_API_KEY")

    client = RunningHubClient(api_key=api_key)
    runner = AiAppRunner(client)

    uploaded_assets: list[str] = []
    for file_path in args.file:
        upload_result = client.upload_file(file_path)
        file_name = ((upload_result.get("data") or {}).get("fileName"))
        if not file_name:
            raise SystemExit(f"上传失败：{upload_result}")
        uploaded_assets.append(file_name)

    overrides = build_app_overrides(
        args.app_id,
        uploaded_assets=uploaded_assets,
        prompt=args.prompt,
    )
    result = runner.run_and_wait(
        args.app_id,
        node_overrides=overrides,
        poll_interval=args.poll_interval,
        timeout_seconds=args.timeout_seconds,
    )

    output = json.dumps(result, ensure_ascii=False, indent=2)
    print(output)
    if args.save_json:
        Path(args.save_json).write_text(output + "\n", encoding="utf-8")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: skills/runninghub-api-dev/assets/sdk/node/examples/run-app.mjs
import { mkdir, writeFile } from 'node:fs/promises';
import path from 'node:path';

import { APP_PRESETS, buildAppOverrides } from '../app-presets.mjs';
import { AiAppRunner } from '../quick-create-client.mjs';
import { RunningHubClient } from '../runninghub-client.mjs';

function parseArgs(argv) {
  const args = {
    files: [],
    prompt: null,
    saveJson: null,
    pollIntervalMs: 12_000,
    timeoutMs: 1_800_000,
  };

  for (let index = 0; index < argv.length; index += 1) {
    const token = argv[index];
    if (token === '--app-id') {
      args.appId = argv[index + 1];
      index += 1;
      continue;
    }
    if (token === '--file') {
      args.files.push(argv[index + 1]);
      index += 1;
      continue;
    }
    if (token === '--prompt') {
      args.prompt = argv[index + 1];
      index += 1;
      continue;
    }
    if (token === '--save-json') {
      args.saveJson = argv[index + 1];
      index += 1;
      continue;
    }
    if (token === '--poll-interval-ms') {
      args.pollIntervalMs = Number(argv[index + 1]);
      index += 1;
      continue;
    }
    if (token === '--timeout-ms') {
      args.timeoutMs = Number(argv[index + 1]);
      index += 1;
      continue;
    }
    if (token === '--help' || token === '-h') {
      args.help = true;
      continue;
    }
    throw new Error(`未知参数：${token}`);
  }
  return args;
}

function printHelp() {
  console.log(`运行 RunningHub AI 应用示例

用法：
  node run-app.mjs --app-id <id> --file <path> [--file <path> ...]

参数：
  --app-id             必填，四个预置应用之一
  --file               可重复传入，本地输入文件
  --prompt             覆盖默认提示词
  --save-json          保存执行结果到 JSON 文件
  --poll-interval-ms   轮询间隔，默认 12000
  --timeout-ms         超时时间，默认 1800000
`);
}

async function main() {
  const args = parseArgs(process.argv.slice(2));
  if (args.help) {
    printHelp();
    return 0;
  }
  if (!args.appId || !APP_PRESETS[args.appId]) {
    throw new Error(`缺少或不支持的 --app-id，可选值：${Object.keys(APP_PRESETS).join(', ')}`);
  }
  const apiKey = process.env.RUNNINGHUB_API_KEY;
  if (!apiKey) {
    throw new Error('缺少环境变量 RUNNINGHUB_API_KEY');
  }

  const client = new RunningHubClient({ apiKey });
  const runner = new AiAppRunner(client);

  const uploadedAssets = [];
  for (const filePath of args.files) {
    const uploadResult = await client.uploadFile(filePath);
    const fileName = uploadResult?.data?.fileName;
    if (!fileName) {
      throw new Error(`上传失败：${JSON.stringify(uploadResult)}`);
    }
    uploadedAssets.push(fileName);
  }

  const overrides = buildAppOverrides(args.appId, {
    uploadedAssets,
    prompt: args.prompt,
  });
  const result = await runner.runAndWait(args.appId, {
    nodeOverrides: overrides,
    pollIntervalMs: args.pollIntervalMs,
    timeoutMs: args.timeoutMs,
  });

  const output = `${JSON.stringify(result, null, 2)}\n`;
  process.stdout.write(output);
  if (args.saveJson) {
    await mkdir(path.dirname(args.saveJson), { recursive: true });
    await writeFile(args.saveJson, output, 'utf8');
  }
  return 0;
}

main().then(
  (code) => {
    process.exitCode = code;
  },
  (error) => {
    console.error(error?.stack ?? String(error));
    process.exitCode = 1;
  },
);
